src/controllers/expenses.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const { postExpense,
+  getExpenseById,
+  getExpenses,
+  deleteExpense,
+  updateExpense } = require('../services/expenses');
+const { getUserById } = require('../services/users');
+const { allUsers } = require('../controllers/users');
+let expenses = [];
+
+const clearExpensesArray = () => {
+  expenses = [];
+};
+const controllerPostExpense = (req, res) => {
+  const { userId } = req.body;
+
+  const user = getUserById(+userId, allUsers());
+  const expense = postExpense(req.body, expenses);
+
+  if (user === null) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  res.statusCode = 201;
+  res.send(expense);
+};
+
+const controllerGetExpenses = (req, res) => {
+  const normalizedURL = new URL(req.url, `http://${req.headers.host}`);
```

Comment:
Don’t need to use URL constructor. Try to console req, you have query object there